### PR TITLE
feat: supports lsp 0.8.+ contracts

### PR DESCRIPTION
**Pull request by samuel-videau**
https://app.clickup.com/t/860q209t3

### CODE REVIEW

File: utils/web3.ts [hunk 1]
```diff
@@ -285,7 +283,7 @@ export const checkInterface = async (address: string, web3: Web3) => {
     isErc1271,
     isLsp0Erc725Account,
     isLsp1UniversalReceiver,
-    isLsp1UniversalReceiverDelegate,
+    // isLsp1UniversalReceiverDelegate,
```
Comment (magalimorin18) [inline utils/web3.ts:286]: FMI, why is this one commented and not just removed ? Do we want to activate it again later ?

**Review by magalimorin18 — COMMENTED** [review]
lgtm

File: utils/web3.ts (hunk 1 above)
Comment (samuel-videau) [reply utils/web3.ts:286]: Ahh it's a miss from me, let me change that thx

File: utils/web3.ts [hunk 2]
```diff
@@ -217,25 +219,21 @@ export const checkInterface = async (address: string, web3: Web3) => {
   let isLsp0Erc725Account = false;
   try {
     isLsp0Erc725Account = await Contract.methods
-      .supportsInterface(LSP_INTERFACE_IDS.LSP0ERC725Account)
+      .supportsInterface(INTERFACE_IDS.LSP0ERC725Account)
       .call();
+    if (!isLsp0Erc725Account) {
+      isLsp0Erc725Account = await Contract.methods
+        .supportsInterface(LEGACY_LSP0_ACCOUNT_INTERFACE_ID)
```
Comment (CallumGrindle) [inline utils/web3.ts:226]: we might just want to drop support of old contracts

**Review by CallumGrindle — APPROVED** [review]
lgtm

File: utils/web3.ts (hunk 2 above)
Comment (Hugoo) [reply utils/web3.ts:226]: Yeah maybe once we get rid of L14 we can remove these legacy contracts. It can be useful to have it for debugging.

**Pull request merged**